x/mongo/driver/errors.go

```diff
@@ -317,12 +322,14 @@ func (e Error) NotMaster() bool {
 			return true
 		}
 	}
-	return strings.Contains(e.Message, "not master")
+	noCode := e.Code == 0
+	return noCode && strings.Contains(e.Message, "not master")
 }
 
 // NamespaceNotFound returns true if this errors is a NamespaceNotFound error.
 func (e Error) NamespaceNotFound() bool {
-	return e.Code == 26 || e.Message == "ns not found"
+	noCode := e.Code == 0
+	return e.Code == 26 || noCode && e.Message == "ns not found"
```

Comment:
add parentheses around `(noCode && e.Message == "ns not found")` to make the grouping more obvious

Reply:
Sure! Added the same grouping for `IsMaxTimeMSExpiredError()`, as well.